Obsidian.API/_Types/Painting.cs

```diff
@@ -6,14 +6,14 @@
 
 namespace Obsidian.API
 {
-    public class Painting
+    public struct Painting
     {
 
-        public readonly short Id;
-        public readonly byte X;
-        public readonly byte Y;
-        public readonly byte Width;
-        public readonly byte Height;
+        public short Id { get; private set; }
```

Comment:
Properties can be set in the constructor no matter if the `set` is missing, I would suggest to remove it.

Reply:
And then even make the struct `readonly`.